### ISSUE TITLE
Unique check on pretty urls is not considering sites

### ISSUE THREAD

**Issue by roland4432** [pimcore/pimcore#7219]
# Bug Report

When saving documents the pretty url is checked to be unique. If not a warning will be prompted. The check doesn't take into account, that the check doesn't need to be global, but per site, since the combination of domain + pretty url has to be unique only.

### Expected behavior

The unique check when storing documents with pretty urls should check for pretty urls per site.

### Actual behavior

The check goes through all documents to check if the pretty url is unique.

### Steps to reproduce

Create two sites and add a document to each of them. Enter the same pretty url for the created documents and hit save.
Second document will bring the warning that this pretty url is already taken (but still saves the document)

**Comment by mrpdean** [comment]
Just ran into this issue myself.  Is there any workaround for it?
Thanks

**Comment by roland4432** [comment]
As far as I remember the document with the newly set pretty url will be saved and routing works fine.
Maybe I can come up with a PR next week. Seems to be a rather uncommon usecase, that people use multisites. since there was not much feedback regarding this issue.

**Comment by mrpdean** [comment]
No, the document with the newly set pretty url won't save.  Instead you get this error: [URL path already exists](https://github.com/pimcore/pimcore/blob/21d29134ef366835ed5acaf4cb472c4efa518624/bundles/AdminBundle/Controller/Admin/Document/PageController.php#L358)

However, I was able to restructure things so that I don't need the pretty urls anymore so it's no longer a problem for me at least. Thanks

**Comment by roland4432** [comment]
IMHO still a problem, that needs to be fixed.

**Comment by NiklasBr** [comment]
This is still relevant

**Comment by NiklasBr** [comment]
This is still important

**Comment by NiklasBr** [comment]
This is still relevant

**Comment by scrummer** [comment]
@roland4432 Just stumbled across this and noticed that this issue has been resolved in #14130 so I guess this issue can be closed :)

**Comment by brusch** [comment]
@scrummer thanks for the hint! 👍

**Comment by twin-elements** [comment]
The problem has not been solved. (Pimcore 11.5.3)

I can create 2 same prettyUrl but then there are cases when wrong document is downloaded.

![Image](https://github.com/user-attachments/assets/09240da6-6cd7-48aa-bf75-f6184ac6acf3)

The search is not dependent on the site ID, so there is no control as to the result.
![Image](https://github.com/user-attachments/assets/b0853786-431a-4f6f-959e-2455d3623d81)

ID 52 is added for the subdomain
![Image](https://github.com/user-attachments/assets/5500431b-01fd-4e63-ad3a-97288fec284c)

ID 53 is added for the main domain
![Image](https://github.com/user-attachments/assets/0400773c-062a-4c9a-adf7-cfd483b31d18)

In summary, if the document is added for the subdomain first and then for the domain then an error will appear.

But if I add the document first for the main domain and then for the suddomain, everything works fine.
However, it is an error, because the ordinary user of the site will not even know about it.
